modules/mod-deep-learning/DeepLearningEffect.cpp

```diff
@@ -0,0 +1,182 @@
+/**********************************************************************
+
+   Audacity: A Digital Audio Editor
+   Audacity(R) is copyright (c) 1999-2021 Audacity Team.
+
+   DeepLearningEffect.cpp
+   Hugo Flores Garcia
+
+******************************************************************/
+/**
+
+*/
+/*******************************************************************/
+
+#include "DeepLearningEffect.h"
+
+#include <wx/stattext.h>
+
+#include "FileNames.h"
+#include "Shuttle.h"
+#include "ShuttleGui.h"
+#include "WaveTrack.h"
+
+#include "effects/LoadEffects.h"
+
+#include <torch/script.h>
+
+// DeepLearningEffectBase implementation
+
+std::string DeepLearningEffect::GetDeepEffectID()
+{ return "waveform-to-waveform";}
+
+const ComponentInterfaceSymbol DeepLearningEffect::Symbol
+{ XO("Deep Learning Effect") };
+
+// register source separation
+namespace{ BuiltinEffectsModule::Registration< DeepLearningEffect > reg; }
+
+DeepLearningEffect::DeepLearningEffect()
+{
+   SetLinearEffectFlag(false);
+}
+
+DeepLearningEffect::~DeepLearningEffect()
+{
+}
+
+// ComponentInterface implementation
+
+ComponentInterfaceSymbol DeepLearningEffect::GetSymbol()
+{
+   return Symbol;
+}
+
+TranslatableString DeepLearningEffect::GetDescription()
+{
+   return XO("The goal of audio source separation is to isolate "
+             "the sound sources in a given mixture of sounds.");
```

Comment:
CMakeLists.txt suggests that this effect has other uses besides source separation.